### ISSUE TITLE
Which branch was missed?

### ISSUE THREAD

**Issue by pjlewis** [jacoco/jacoco#87]
I'm switching a project from Cobertura to JaCoCo - as part of an eventual upgrade to Java 1.7.  I've noticed that JaCoCo indicates missed branches but it doesn't tell you which branch was missed.  So in code like the following...

```
if(a==b || c==d) { ... }
```

JaCoCo might say '1 of 4 branches' missed.  But it doesn't tell me which branch (or even which term in the condition).  Cobertura used to give an indication in the form '%50/%100' (I think) which gave some idea of where to start.  Could something like this be added?

**Comment by rliesenfeld** [comment]
This is definitely an interesting feature to have in a coverage tool. I have it working reasonably well in my own tool (JMockit Coverage), so it's doable (and yes, it does require parsing the source code). Based on my experience, I can say it's quite difficult to get right for all possible cases; it will probably take me several years still.

**Comment by marchof** [comment]
@rliesenfeld Thanks for sharing your insights here! That's why I don't think we will ever do source code parsing for JaCoCo. Rather we use compiler features like the CharacterRangeTable attribute.

**Comment by C-Otto** [comment]
My example is simple.

```
 L1
    LINENUMBER 47 L1
    GETSTATIC Foo.lastUsage : Ljava/lang/Class;
    IFNULL L2
    ALOAD 0
    GETSTATIC FoolastUsage : Ljava/lang/Class;
    IF_ACMPEQ L2
L3
    ....
L2
    ....
```

I assume JaCoCo knows that IFNULL and IF_ACMPEQ are part of the same line, coming up with 2^2=4 branches. However, just looking at the Bytecode, you can also see that there are only three branches:

1. IFNULL true, jumping to L2
2. IFNULL false and IF_ACMPEQ true, jumping to L2
3. IFNULL false and IF_ACMPEQ false, continuing in L3

Everything else is not possible, as the code between IFNULL and IF_ACMPEQ has no label. I don't see an issue detecting this from looking at the bytecode alone. Please enlighten me.

**Comment by marchof** [comment]
@C-Otto This issue is about creating a mapping back to the source code to give the user meaningful information about individual branches in from the source point of view. I do not understand how your example relates to this.

**Comment by C-Otto** [comment]
@marchof you're right, I'm sorry. I misread the issue. Furthermore, I think the issue I described is already fixed in master. Thanks!

**Comment by bmattoso** [comment]
Hi,

Did I find this PR #317 and have you a prevision to release this feature?
Thanks!

**Comment by caiohcl** [comment]
Is there any ETA to release this feature?
